hedgehog/src/Hedgehog/Internal/State.hs

```diff
@@ -402,11 +401,26 @@ data Command gen m (state :: (Type -> Type) -> Type) =
         [Callback input output state]
     }
 
--- | Checks that input for a command can be executed in the given state.
+
+-- | An actionable command is a command that can be turned into
+--   an action by running its generator.
+--
```

Comment:
Perhaps add a note that we only construct actions where `inputGen state0` was `Just`, matching the new invariant.